tasks/MatchedCatalogsAnalysis.py

```diff
@@ -43,3 +42,36 @@ def __init__(self, config, *args, **kwargs):
 
     def run(self, matchedCatalog):
         return self.measure.run(matchedCatalog, self.config.connections.metric)
+
+
+class MatchedMultiCatalogAnalysisTaskConnections(MetricConnections,
+                                                 dimensions=("tract", "patch", "abstract_filter",
+                                                 "instrument", "skymap")):
+    matchedCatalogMulti = pipeBase.connectionTypes.Input(doc="Input matched catalog.",
+                                                    dimensions=("tract", "patch", "instrument"),
+                                                    storageClass="SimpleCatalog",
+                                                    name="matchedCatalogMulti")
+    measurement = pipeBase.connectionTypes.Output(doc="Resulting matched catalog.",
+                                                  dimensions=("tract", "patch",
+                                                              "instrument","abstract_filter"),
+                                                  storageClass="MetricValue",
+                                                  name="metricvalue_{package}_{metric}")
+
+
+class MatchedMultiCatalogAnalysisTaskConfig(MetricConfig,
+                               pipelineConnections=MatchedMultiCatalogAnalysisTaskConnections):
+    measure = pexConfig.ConfigurableField(
+        target=NumSourcesTask,  # This is the simplest measurement I can think of
+        doc="Measure task")
+
+
+class MatchedMultiCatalogAnalysisTask(MetricTask):
```

Comment:
As before, I don't think we need to duplicate the run code:
```python
class MatchedMultiCatalogAnalysisTask(MatchedCatalogAnalysisTask):
    ConfigClass = MatchedMultiCatalogAnalysisTaskConfig
    _DefaultName = "matchedMultiCatalogAnalysisTask"

Reply:
Actually, after trying this, I think it's worth duplicating the run code so we can have a `matchedCatalogMulti` object that is different from the `matchedCatalog`.

Reply:
I'm confused by this.  You will still have a separate class for the multi part.  It doesn't require duplicating the run code for that to happen.  If you are concerned that one inherits from the other, we should probably break out an abstract base class that they both inherit from.

Reply:
I'm a bit confused, too. In the connections class for `MatchedMultiCatalogAnalysisTask`, the input dataset is called `matchedCatalogMulti`. When I set it up to inherit from the other run method, it choked because `measureAB1` (and, I think, the connections class itself) is looking for `matchedCatalogMulti`, but the run method returns `matchedCatalog`. Maybe there's a way to reconcile this other than re-implementing the same run method with different outputs, but I'm not sure how to do it.

Reply:
OK.  You can leave it as is and we'll look into cleaning it up when we refactor the contexts.